src/blockchain/lending/market.ts

```diff
@@ -0,0 +1,237 @@
+ * @param address optional address - checks current signer if not provided
+ * @returns
+ */
+export const getBorrowingLimit = async (
+  marketName: Markets,
+  provider: BaseProvider,
+  address?: string
+) => {
+  const markets = await getMarkets(provider)
+  const market = markets[marketName]
+  if (!address) {
+    address = await getSignerAddress(provider)
+  }
+  return market.borrowingLimit(address)
+}
+
+// cSpell:disable
+
+// -- Approval functions for collateralization of each xAsset ---
+
+export const approveXAAVEa = async (
+  amount: BigNumber,
+  provider: BaseProvider
+) => {
+  const network = await provider.getNetwork()
+  const contract = new Contract(
+    ADDRESSES['X_AAVE_A'][network.chainId],
+    Abi.ERC20,
+    provider
+  )
+
+  return contract.approve(
+    ADDRESSES[LENDING_LIQUIDITY_POOL][network.chainId],
+    amount
+  )
+}
+
+export const approveXAAVEb = async (
+  amount: BigNumber,
+  provider: BaseProvider
+) => {
+  const network = await provider.getNetwork()
+  const contract = new Contract(
+    ADDRESSES['X_AAVE_B'][network.chainId],
+    Abi.ERC20,
+    provider
+  )
+
+  return contract.approve(
+    ADDRESSES[LENDING_LIQUIDITY_POOL][network.chainId],
+    amount
+  )
+}
+
+export const approveXINCHa = async (
+  amount: BigNumber,
+  provider: BaseProvider
+) => {
+  const network = await provider.getNetwork()
+  const contract = new Contract(
+    ADDRESSES['X_INCH_A'][network.chainId],
+    Abi.ERC20,
+    provider
+  )
+
+  return contract.approve(
+    ADDRESSES[LENDING_LIQUIDITY_POOL][network.chainId],
+    amount
+  )
+}
+
+export const approveXINCHb = async (
+  amount: BigNumber,
+  provider: BaseProvider
+) => {
+  const network = await provider.getNetwork()
+  const contract = new Contract(
+    ADDRESSES['X_INCH_B'][network.chainId],
+    Abi.ERC20,
+    provider
+  )
+
+  return contract.approve(
+    ADDRESSES[LENDING_LIQUIDITY_POOL][network.chainId],
+    amount
+  )
+}
+
+export const approveXKNCa = async (
+  amount: BigNumber,
+  provider: BaseProvider
+) => {
+  const network = await provider.getNetwork()
+  const contract = new Contract(
+    ADDRESSES['X_KNC_A'][network.chainId],
+    Abi.ERC20,
+    provider
+  )
+
+  return contract.approve(
+    ADDRESSES[LENDING_LIQUIDITY_POOL][network.chainId],
+    amount
+  )
+}
+
+export const approveXKNCb = async (
+  amount: BigNumber,
+  provider: BaseProvider
+) => {
+  const network = await provider.getNetwork()
+  const contract = new Contract(
+    ADDRESSES['X_KNC_B'][network.chainId],
+    Abi.ERC20,
+    provider
+  )
+
+  return contract.approve(
+    ADDRESSES[LENDING_LIQUIDITY_POOL][network.chainId],
+    amount
+  )
+}
+
+const getApprovedAmountXAsset = async (
+  marketName: Markets,
+  address: string,
+  provider: BaseProvider
+) => {
+  const network = await provider.getNetwork()
+  let marketAddress
+  switch (marketName) {
+    case Markets.xAAVEaMarket:
+      marketAddress = ADDRESSES['X_AAVE_A'][network.chainId]
```

Comment:
What's the significance of `'X_AAVE_A'` string here? Shouldn't we be using the constant `X_AAVE_A` imported from `abis` instead of the string literal?